etcdserver/raft.go

```diff
@@ -254,18 +255,19 @@ func restartNode(cfg *ServerConfig, snapshot *raftpb.Snapshot) (types.ID, raft.N
 		MaxSizePerMsg:   maxSizePerMsg,
 		MaxInflightMsgs: maxInflightMsgs,
 	}
-	n := raft.RestartNode(c)
+	n = raft.RestartNode(c)
 	raftStatus = n.Status
-	return id, n, s, w
+	return
```

Comment:
```
Naked returns are okay if the function is a handful of lines. Once it's a medium-sized function, be explicit with your return values. Corollary: it's not worth it to name result parameters just because it enables you to use naked returns. Clarity of docs is always more important than saving a line or two in your function.

Finally, in some cases you need to name a result parameter in order to change it in a deferred closure. That is always okay.
```